apps/emqx/src/emqx_external_broker.erl

```diff
@@ -0,0 +1,149 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2024 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+-module(emqx_external_broker).
+
+-callback forward(emqx_router:external_dest(), emqx_types:delivery()) ->
+    emqx_types:deliver_result().
+
+-callback should_route_to_external_dests(emqx_types:message()) -> boolean().
+
+-callback maybe_add_route(emqx_types:topic()) -> ok.
+-callback maybe_delete_route(emqx_types:topic()) -> ok.
+
+-callback maybe_add_shared_route(emqx_types:topic(), emqx_types:group()) -> ok.
+-callback maybe_delete_shared_route(emqx_types:topic(), emqx_types:group()) -> ok.
+
+-callback maybe_add_persistent_route(emqx_types:topic(), emqx_persistent_session_ds:id()) -> ok.
+-callback maybe_delete_persistent_route(emqx_types:topic(), emqx_persistent_session_ds:id()) -> ok.
```

Comment:
Nit. From the point of view of the user of a CBM implementing this behavior, it's sort of weird to see `maybe_` here. These are essentially "event handlers", and the CBM is free to do or not to do anything with some route according to its internal logic. I.e. if `maybe_add_route/1` was named just `add_route/1`, it's not like we expect the CBM is _required_ to change some initial state.

Reply:
Agree, fixed